docs/src/js/main.js

```diff
@@ -248,6 +248,20 @@ document.querySelectorAll('.progress-bar-reset').forEach(function(el) {
     });
 });
 
+// STAR RATING SELECT
+const starRadios =  document.querySelectorAll('.star-rating-select input[name="radio-star-rating-select"]');
+
+starRadios.forEach((radio, index) => {
+    radio.addEventListener('change', function () {
+        const starSelectedClass = 'star-rating-select_filled';
```

Comment:
This needs to be `--filled`

On the same note, in order to have a modifier on an element, it should have a root class.
For example, I think the root class should be `star-rating__control` (I'm open), and then the modifier `star-rating__control--filled` should be applied if it's filled. 
From what I see it seems like the styles are being applied to `input[type="radio"]`. I think it might be better to do:
`.star-rating .star-rating__control { //apply styles for input[type="radio" }`
`.star-rating .star-rating__control.star-rating__control--filled { //apply styles for filled }`
That's what I'm thinking. I'm happy to discuss or if Ian has any other opinions. 

I know it's a lot, but it's worth getting it right imo, so it saves us in the longrun.

Reply:
Agreed. Taking additional time to get the BEM right is important, even if it means a little back and forth in the PR. Then we should never (or rarely) have to come back to it again :-)

Reply:
Good spot btw. I neglected to look at the class being added in the javascript.